Client/Telemetry/UnifiedTelemetry.swift

```diff
@@ -0,0 +1,95 @@
+}
+
+class TelemetryEventObject {
+    public static let app = "app"
+//    public static let searchBar = "search_bar"
+//    public static let eraseButton = "erase_button"
+//    public static let settingsButton = "settings_button"
+//    public static let setting = "setting"
+//    public static let menu = "menu"
+//    public static let pasteAndGo = "paste_and_go"
+}
+
+class UnifiedTelemetry {
+
+    init(profile: Profile) {
+        NotificationCenter.default.addObserver(self, selector: #selector(UnifiedTelemetry.appWillResignActive(notification:)), name: NSNotification.Name.UIApplicationWillResignActive, object: nil)
+
+        NotificationCenter.default.addObserver(self, selector: #selector(UnifiedTelemetry.appDidEnterBackground(notification:)), name: NSNotification.Name.UIApplicationDidEnterBackground, object: nil)
+
+        NotificationCenter.default.addObserver(self, selector: #selector(UnifiedTelemetry.appDidBecomeActive(notification:)), name: NSNotification.Name.UIApplicationDidBecomeActive, object: nil)
+
+        setupCorePing(withProfile: profile)
+    }
+
+    @objc func appWillResignActive(notification: NSNotification) {
+        Telemetry.default.recordSessionEnd()
+    }
+
+    @objc func appDidEnterBackground(notification: NSNotification) {
+        Telemetry.default.queue(pingType: CorePingBuilder.PingType)
+        Telemetry.default.scheduleUpload(pingType: CorePingBuilder.PingType)
+    }
+
+    @objc func appDidBecomeActive(notification: NSNotification) {
+        Telemetry.default.recordSessionStart()
+        Telemetry.default.recordEvent(category: TelemetryEventCategory.action, method: TelemetryEventMethod.foreground, object: TelemetryEventObject.app)
+    }
+
+    fileprivate func setupCorePing(withProfile profile: Profile) {
+        let telemetryConfig = Telemetry.default.configuration
+        telemetryConfig.appName = AppInfo.displayName
+        telemetryConfig.userDefaultsSuiteName = AppInfo.sharedContainerIdentifier
+        telemetryConfig.dataDirectory = .documentDirectory
+
+        #if DEBUG
+            telemetryConfig.updateChannel = "debug"
+            telemetryConfig.isCollectionEnabled = false
+            telemetryConfig.isUploadEnabled = false
+        #else
+            telemetryConfig.updateChannel = "release"
+            let sendUsageData = profile.prefs.boolForKey(AppConstants.PrefSendUsageData) ?? true
+            telemetryConfig.isCollectionEnabled = sendUsageData
+            telemetryConfig.isUploadEnabled = sendUsageData
+        #endif
+
+
+        // TODO: Add these items at ping root
+//        if let newTabChoice = profile.prefs.stringForKey(NewTabAccessors.PrefKey) {
+//            out["defaultNewTabExperience"] = newTabChoice as AnyObject?
+//        }
+//
+//        if let chosenEmailClient = profile.prefs.stringForKey(PrefsKeys.KeyMailToOption) {
+//            out["defaultMailClient"] = chosenEmailClient as AnyObject?
+//        }
+
+        // TODO: Add pref items
```

Comment:
I wonder if we can add some now, or do we wait?

Reply:
Yes, you can add these types of measurements here. For reference, here's how Focus inits telemetry-ios:

https://github.com/mozilla-mobile/focus-ios/blob/master/Blockzilla/AppDelegate.swift#L23-L59

Reply:
Also leave out for first landing?